cirq/sim/__init__.py

```diff
@@ -17,8 +17,6 @@
 from cirq.sim.density_matrix_utils import (
     measure_density_matrix,
     sample_density_matrix,
```

Comment:
We may need to import the names into `cirq.sim` as well to avoid breaking clients that depend on this.

Reply:
There have been several comments on #2810 agreeing with creating `cirq.qis`. Shall I go ahead and move everything to `qis/states.py`? Also I was hoping there would be a better way to deprecate importing from old modules, but if there isn't then sure, we can just import into them.

Reply:
Yes, let's go ahead with `cirq.qis`. Thanks!

I'm not aware of a better way. I think you should move the code to the new location, import into the old location and add the deprecated decorator.

Reply:
The deprecated decorator goes over the function though. As far as I can tell, there is no way to mark _the act of importing from the old module_ as deprecated.

Reply:
Decorator is just a function (that takes and returns functions). Also, you don't need to mark the act of importing as deprecated. Instead, think about a symbol under `cirq.sim` as being defined using the `deprecated` function and the new symbol under `cirq.qis`.

This may work:

In a module inside the `cirq.qis` package:

```python
def von_neumann_entropy(...):
    ...
```

In a module inside the `cirq.sim` package:

```python
from cirq.qis import von_neumann_entropy as qis_von_neumann_entropy

von_neumann_entropy = deprecated(qis_von_neumann_entropy, ...)
```

Reply:
Thanks, that gave a way to deprecate the functions. However, this doesn't let us deprecate `STATE_VECTOR_LIKE`. I just left that in the old file.

Reply:
In order to deprecate `one_hot` and `eye_tensor` in `linalg.states`, I need to import `qis` in that file. But since `qis` depends on `linalg`, this causes a circular import, which is causing the MacOS test to fail.